tests/ndc/replication_task_batching_test.go

```diff
@@ -0,0 +1,337 @@
+	s.Assertions = require.New(s.T())
+	s.ProtoAssertions = protorequire.New(s.T())
+}
+
+func (s *NDCReplicationTaskBatchingTestSuite) TestHistoryReplicationTaskAndThenRetrieve() {
+	versions := []int64{1, 1, 21, 31, 301, 401, 601, 501, 801, 1001, 901, 701, 1101}
+	executions := make(map[workflow.Execution][]*historypb.History)
+	for _, version := range versions {
+		workflowID := "replication-message-test" + uuid.New()
+		runID := uuid.New()
+		var historyBatch []*historypb.History
+		s.generator = test.InitializeHistoryEventGenerator(s.namespace, s.namespaceID, version)
+		for s.generator.HasNextVertex() {
+			events := s.generator.GetNextVertices()
+
+			historyEvents := &historypb.History{}
+			for _, event := range events {
+				historyEvents.Events = append(historyEvents.Events, event.GetData().(*historypb.HistoryEvent))
+			}
+			historyBatch = append(historyBatch, historyEvents)
+
+			s.standByReplicationTasksChan <- s.createHistoryEventReplicationTaskFromHistoryEventBatch( // supply history replication task one by one
+				s.namespaceID.String(),
+				workflowID,
+				runID,
+				historyEvents.Events,
+				nil,
+				s.eventBatchesToVersionHistory(nil, historyBatch).Items,
+			)
+		}
+		execution := workflow.Execution{
+			ID:    workflowID,
+			RunID: runID,
+		}
+		executions[execution] = historyBatch
+	}
+	time.Sleep(5 * time.Second) // 5 seconds is enough for the history replication task to be processed and applied to passive cluster
+
+	for execution, historyBatch := range executions {
+		s.assertHistoryEvents(context.Background(), s.namespaceID.String(), execution, historyBatch)
+	}
+}
+
+func (s *NDCReplicationTaskBatchingTestSuite) assertHistoryEvents(
+	ctx context.Context,
+	namespaceId string,
+	execution workflow.Execution,
+	historyBatch []*historypb.History,
+) {
+	mockClientBean := client.NewMockBean(s.controller)
+	mockClientBean.
+		EXPECT().
+		GetRemoteAdminClient(s.passiveClusterName).
+		Return(s.passtiveCluster.GetAdminClient(), nil).
+		AnyTimes()
+
+	serializer := serialization.NewSerializer()
+	passiveClusterFetcher := replication.NewHistoryPaginatedFetcher(
+		nil,
+		mockClientBean,
+		serializer,
+		nil,
+		s.logger,
+	)
+
+	passiveIterator := passiveClusterFetcher.GetSingleWorkflowHistoryPaginatedIterator(
+		ctx, s.passiveClusterName, namespace.ID(namespaceId), execution.ID, execution.RunID, 0, 1, 0, 0)
+
+	index := 0
+	for passiveIterator.HasNext() {
+		s.True(passiveIterator.HasNext())
+		passiveBatch, err := passiveIterator.Next()
+		s.NoError(err)
+		inputEvents := historyBatch[index].Events
+		index++
+		inputBatch, _ := s.serializer.SerializeEvents(inputEvents, enumspb.ENCODING_TYPE_PROTO3)
+		s.Equal(inputBatch, passiveBatch.RawEventBatch)
+	}
+}
+
+func (s *NDCReplicationTaskBatchingTestSuite) registerNamespace() {
+	s.namespace = namespace.Name("test-simple-workflow-ndc-" + common.GenerateRandomString(5))
```

Comment:
this namespace will have no AccountID. It will be "namespaceName" not "namespaceName.AccountID". Want to ensure this is what was meant, because it's nonstandard

Reply:
`namespace.accountId` convention is used inside temporal.io but it should not be a requirement to run the temporal server.